### PR TITLE
Add facility to deepcopy bound methods in Py2.6 and apply to grains

### PR DESCRIPTION
**Pull request by cachedout**
In #28587, we failed to account for the inability of Py2.6 to deepcopy bound methods. This provides a workaround for that.

### CODE REVIEW

**Comment by cachedout** [comment]
Go Go Jenkins!

File: salt/utils/compat.py [hunk 1]
```diff
@@ -20,3 +22,27 @@ def pack_dunder(name):
...
+    '''
+    Compatibility helper function to allow copy.deepcopy copy bound methods
+    which is broken on Python 2.6, due to the following bug:
+    https://bugs.python.org/issue1515
+
+    Warnings:
+        - This method will mutate the global deepcopy dispatcher, which means that
+        this function is NOT threadsafe!
+
+        - Not Py3 compatable. The intended use case is deepcopy compat for Py2.6
```
Comment (s0undt3ch) [inline salt/utils/compat.py]: Compatible? :smile:

**Comment by s0undt3ch** [comment]
:evil:

**Comment by cachedout** [comment]
Yeah, it's a wee bit evil. I won't lie. :[

**Pull request merged**